carbon-design-system/carbon
# feat(Tabs): Dismissable tabs

Closes #12813 

This adds dismissable functionality to tabs

- Adds new `dismissable` & `onTabCloseRequest` props to `Tabs`
- Adds new `Dismissable` Tabs story
- Adds Dismissable tabs docs guidance to Tabs story
- Adds VRT & RTL tests for dismissable tabs

TODO:
- [x] Confirm disabled state
- [x] Confirm hover state


#### Changelog

**New**

- `dismissable` & `onTabCloseRequest` props in `Tabs`, also passed through the `TabsContext`
- dismissable VRT story
- RTL tests for dismissable tabs
- Tabs docs guidance for dismissable tabs
- Dismissable tab story
- close button styling rule for dismissable tab
- `Dismissable` control in `Tabs` Playground story

**Changed**

- Refactored `Tabs` scroll buttons visibility logic to account for deletable tabs
- Datatable docs package import correction (use `@carbon/react` instead of `carbon-components-react`)
- Overrider renderIcon with Dismiss(close) Icon on `Tab` when `TabsContext`'s value has `dismissable`
- `onKeyDown` handle in `Tab` to call `onTabCloseRequest` when `delete` is pressed on dismissable tab

#### Testing / Reviewing

1. https://deploy-preview-13529--carbon-components-react.netlify.app/?path=/story/components-tabs--dismissable
- Verify styling matches spec
![image](https://user-images.githubusercontent.com/40550942/233246536-79b5ae0b-13de-46fa-8eab-bd2bbc092f72.png)
- Verify dismissable behavior
- Verify reset behavior
- Verify keyboard functionality

2. Verify other tab stories still work/behave as expected and no visual/behavioral regressions
3. Verify Dismissable docs are coherent and readable https://deploy-preview-13529--carbon-components-react.netlify.app/?path=/docs/components-tabs--dismissable#dismissable-tabs
